R/sass.R

```diff
@@ -66,11 +64,22 @@ sass <- function(input = NULL, options = sass_options(), output = NULL,
     )
   }
 
-  if (use_cache) {
-    if (file.exists(cache_file)) {
-      # TODO: Set encoding to UTF-8?
-      css <- paste(readLines(cache_file), collapse = "\n")
+  css <- NULL
+  layer <- NULL
+  if (use_cache && file.exists(cache_file)) {
+    # No need to read cache file if output is specified
+    if (!is.null(output)) {
+      fs::file_copy(cache_file, output, overwrite = TRUE)
+      if (isTRUE(write_attachments == FALSE)) {
+        return(invisible())
+      }
+      layer <- extract_layer(input)
+      maybe_write_attachments(layer, output, write_attachments)
+      return(invisible())
     }
+    # If no output is specified, we need to return a character vector
+    # TODO: Set encoding to UTF-8?
+    css <- paste(readLines(cache_file), collapse = "\n")
```

Comment:
Can you either bring in `write_utf8` from shinytest, or use brio here?

Reply:
Let's handle UTF-8 I/O in a follow-up PR